subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -212,7 +212,11 @@ static int json_add_obj_cs(cJSON *parent, const char *str, cJSON *item)
 	if (!parent || !str || !item) {
 		return -EINVAL;
 	}
-	return cJSON_AddItemToObjectCS(parent, str, item) ? 0 : -ENOMEM;
+	if (!cJSON_AddItemToObjectCS(parent, str, item)) {
+		cJSON_Delete(item);
```

Comment:
i would prefer to not delete `item` here, let it be up to the caller to decide what to do with the data.
these static json functions are meant to be simple wrappers for the cJSON functions.  deleting the data changes the behavior.  for example, the existing calls to `json_add_obj_cs` already call `cJSON_Delete` on failure, this change will result in a double delete 💀

Reply:
I agree, see https://github.com/nrfconnect/sdk-nrf/pull/8304#discussion_r938053921 for additional reasoning behind this

Reply:
fixed